### DISCUSSION TITLE
Connecting to a DB using SSL ROOT CERTIFICATE

### DISCUSSION THREAD

**Discussion by fvermelho** [TPC-Council/HammerDB#361] [General]
Guys, I have some PostgreSQL that was provisioned in two differente cloud providers and the access mode for those DB is by using an SSLROOTCERT.
I didn't find where to put the certificate in hammerdb. It's possible to do ?

**Comment by sm-shaw** [comment]
HammerDB includes the option to set the SSL mode: 
```
hammerdb>print dict
Dictionary Settings for PostgreSQL
connection {
 pg_host    = localhost
 pg_port    = 5432
 pg_sslmode = prefer
}
```
as described here: https://www.postgresql.org/docs/current/libpq-ssl.html
The default location for the SSL ROOTCERT is here ~/.postgresql/root.crt or you can set PGSSLROOTCERT

**Reply by fvermelho** [reply]
Thank you. Now I'm able to connect to my instances using the SSL certificate ;-)

**Comment by maddisurendra** [comment]
Hello ,

We are trying to connect to postgresql via hammerdb using ssl certificate, but we are not finding any option on how to pass the certificates in hammerdb?

We dont have postgresql installed locally.

I didn't find where to put the certificate in hammerdb. It's possible to do ?

**Reply by sm-shaw** [reply]
Hi,
This is the same requirement as described in this discussion.  You will have a PostgreSQL client installed rather than a server. 
Follow the details in here  https://www.postgresql.org/docs/current/libpq-ssl.html to configure the correct settings for the client.

**Comment by maddisurendra** [comment]
I have installed postgresql15 on my windows laptop and hammerdb as well.

Not really sure where to pass the 3 certificates , can you please guide me the location where can i pass the SSL certificates so that i can connect via hammerdb.

![dfdsfa](https://user-images.githubusercontent.com/125586858/221330645-31bc7a8e-462c-488e-97eb-7251eab9ac89.JPG)



hammerdb>print dict
Dictionary Settings for PostgreSQL
connection {
 pg_host    = 10.19.15.2
 pg_port    = 5432
 pg_sslmode = prefer
}
tpch       {
 pg_scale_fact         = 1
 pg_tpch_superuser     = 
 pg_tpch_superuserpass = 
 pg_tpch_defaultdbase  = 
 pg_tpch_user          = 
 pg_tpch_pass          = 
 pg_tpch_dbase         = test
 pg_tpch_tspace        = pg_default
 pg_tpch_gpcompat      = false
 pg_tpch_gpcompress    = false
 pg_num_tpch_threads   = 1
 pg_total_querysets    = 1
 pg_raise_query_error  = false
 pg_verbose            = false
 pg_refresh_on         = false
 pg_degree_of_parallel = 25
 pg_update_sets        = 1
 pg_trickle_refresh    = 1000
 pg_refresh_verbose    = false
 pg_cloud_query        = false
 pg_rs_compat          = false
}

**Reply by sm-shaw** [reply]
We are using libpq to connect.  As above this successfully worked on Linux so should do the same on Windows using the correct file i.e.  %APPDATA%\postgresql\root.crt. 

The documentation is here: https://www.postgresql.org/docs/current/libpq-ssl.html look for sections on windows certificates e.g.

To allow server certificate verification, one or more root certificates must be placed in the file ~/.postgresql/root.crt in the user's home directory. **(On Microsoft Windows the file is named %APPDATA%\postgresql\root.crt.)** 

34.19.2. Client Certificates

If the server attempts to verify the identity of the client by requesting the client's leaf certificate, libpq will send the certificate(s) stored in file ~/.postgresql/postgresql.crt in the user's home directory. The certificates must chain to the root certificate trusted by the server. A matching private key file ~/.postgresql/postgresql.key must also be present. 

On Microsoft Windows these files are named **%APPDATA%\postgresql\postgresql.crt** and **%APPDATA%\postgresql\postgresql.key**. The location of the certificate and key files can be overridden by the connection parameters sslcert and sslkey, or by the environment variables PGSSLCERT and PGSSLKEY.

**Comment by maddisurendra** [comment]
Thnq able to connect now

**Reply by sm-shaw** [reply]
Great, thank-you for updating that it works on Windows as well as Linux

**Comment by maddisurendra** [comment]
Do you have the documen

**Comment by maddisurendra** [comment]
Do you have documentation on how to install hammerdb on linux i am facing
issue with display on linux,

**Reply by sm-shaw** [reply]
Lets make this a new discussion topic
